insights/core/__init__.py

```diff
@@ -426,6 +429,159 @@ def _scan(self):
 
         cls.scan(result_key, _scan)
 
+    def get_after(self, timestamp, lines=None, time_format='%Y-%m-%d %H:%M:%S'):
+        """
+        Find all the (available) logs that are after the given time stamp.
+
+        This method should be used by the subclass to provide an actual
+        ``get_after()`` method which provides the standard time stamp format
+        used by its particular log.  This method should only be used if the
+        subclass does not present its own or the caller needs to override
+        the default time format.
+
+        If 'lines' is not supplied, then all lines are used.  Otherwise, the
+        caller can provide a list of lines from a scanner or the ``get()``
+        method.
+
+        This method then finds all lines in that list which have a time stamp
+        after the given timestamp.  Lines that do not contain a time stamp
+        are considered to be part of the previous line and are therefore
+        included if the last log line was included or excluded otherwise.
+
+        This works by converting the time format into a regular expression
+        which matches the time format in the string.  This is then searched
+        for in each line in turn.  Only lines with a time stamp matching this
+        expression will trigger the decision to include or exclude lines.
+        Therefore, if the log for some reason does not contain a time stamp
+        that matches this format, no lines will be returned.
+
+        Lines can be either a single string or a dictionary.  String lines
+        search for a time stamp in the entire line.  Dictionary lines look
+        for a field named 'timestamp', which is (still) searched for a valid
+        time format as above - lines without the 'timestamp' field are
+        treated as continuation lines (see above).
+
+        Note: Some logs - notably /var/log/messages - do not contain a year
+        in the timestamp.  This detected by the absence of a '%y' or '%Y' in
+        the time format.  If that year field is absent, the year is assumed
+        to be the year in the given timestamp being sought.  Some attempt is
+        made to handle logs with a rollover from December to January, by
+        finding when the log's timestamp (with current year assumed) is over
+        eleven months (specifically, 330 days) ahead of or behind the
+        timestamp date and shifting that log date by 365 days so that it is
+        more likely to be in the sought range.  This paragraph is sponsored
+        by syslog.
+
+        No attempt is made to order the lines by date.  This also means that
+        lines can be out of order, as each timestamp is evaluated individually.
+
+        Params:
+            timestamp(datetime.datetime): lines before this time are ignored.
+            lines(list): an optional list of lines from this parser to search.
+            time_format(str): the ``strptime()`` string of the format of the
+            dates in this log.
```

Comment:
@PaulWay your params are not rendered correctly in the output.  The following changes appear to fix it
```
diff --git a/insights/core/__init__.py b/insights/core/__init__.py
index 976e145..47c00cf 100644
--- a/insights/core/__init__.py
+++ b/insights/core/__init__.py
@@ -475,20 +475,20 @@ class LogFileOutput(Parser):
         No attempt is made to order the lines by date.  This also means that
         lines can be out of order, as each timestamp is evaluated individually.
 
-        Params:
+        Parameters:
             timestamp(datetime.datetime): lines before this time are ignored.
             lines(list): an optional list of lines from this parser to search.
             time_format(str): the ``strptime()`` string of the format of the
-            dates in this log.
+                dates in this log.
 
         Yields:
-            (string): a list of lines with timestamps after this date.
+            string: a list of lines with timestamps after this date.
 
         Raises:
             ParseException: If the format conversion string contains a
-            format that we don't recognise.  In particular, no attempt is
-            made to recognise or parse the time zone or other obscure
-            values like day of year or week of year.
+                format that we don't recognise.  In particular, no attempt is
+                made to recognise or parse the time zone or other obscure
+                values like day of year or week of year.
         """
         if lines is None:
             lines = self.lines
```